client/components/happychat/button.jsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import page from 'page';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import viewport from 'lib/viewport';
+import { openChat } from 'state/ui/happychat/actions';
+import Button from 'components/button';
+import Gridicon from 'components/gridicon';
+
+const HappychatButton = React.createClass( {
```

Comment:
`React.createClass` is on a path toward deprecation and we're instead promoting use the `Component` class. With the class, there is no longer autobinding of functions. You can either bind them in a constructor or use Stage 2 class property initializers:

```js
class HappychatButton extends Component {
	constructor() {
		super( ...arguments );
		this.onOpenChat = this.onOpenChat.bind( this );
	}

	// ...
}
```

Or:

```js
class HappychatButton extends Component {
	onOpenChat = () => {
		// ...
	}

	// ...
}
```